### ISSUE TITLE
draw polygon

### ISSUE THREAD

**Issue by AnatolyAG** [openlayers/ol3#6023]
auxiliary shading is drawn from the second point

![issue_openlayer](https://cloud.githubusercontent.com/assets/18345451/19658439/a5380498-9a30-11e6-9527-524cdb1ac37b.png)

**Comment by tchandelle** [comment]
The problem starts at 430aee95fd32d1fe8ec0fbfa28451394c3533806. Like your picture, but with also an error in the console

    Assertion failed: fillStyle or strokeStyle should be defined @ polygonreplay.js:179

Then it's really hard to follow the commit history, on another commit, the error disappear.

Then on another commit, the fill is drawn like this : 
![image](https://cloud.githubusercontent.com/assets/19923657/19676494/ff696c18-9a94-11e6-9ced-a282bb9ad37a.png)

Or sometime like this : 
![image](https://cloud.githubusercontent.com/assets/19923657/19676520/205ef32a-9a95-11e6-866e-ba1e0710b0e3.png)

And finally, back again, like you described it.

**Comment by fredj** [comment]
```
e7bc7121d61d795457cd9d15165c40b2ad8f0cc6 is the first bad commit
commit e7bc7121d61d795457cd9d15165c40b2ad8f0cc6
Author: Andreas Hocevar <andreas.hocevar@gmail.com>
Date:   Tue Sep 13 01:54:53 2016 +0200

    Add option to skip first coordinate for use with closePath()

:040000 040000 325b4191f67e6210bb772c4edf856740230edda6 032b906f7d518af49e04eb5fbe3ec095f0188b21 M      src
:040000 040000 d0bf4231d94ef7f65734f181cd3b4baf88eabd37 8bd2eacbc78a22befa5acfbf0092e355bf76fbce M      test
```

**Comment by fredj** [comment]
The sketch polygon created by the draw interaction is invalid (first coordinate !== last coordinate)

**Comment by ahocevar** [comment]
Should be an easy fix in the Draw interaction to fix that geometry, but I don't have time to work on it this week. Any takers?

**Comment by ahocevar** [comment]
Let's leave this open until #6027 is merged.

**Comment by AnatolyAG** [comment]
Ok.

**Comment by Newborn262** [comment]
So will all this work for me

**Comment by sateeshm** [comment]
is this issue fixed?

**Comment by sateeshm** [comment]
is this issue resolved?

**Comment by ahocevar** [comment]
Yes, it was fixed with #6027.

**Comment by sateeshm** [comment]
Thanks Andreas,


We are using this open layers in my project. For which, when I draw a polygon because of the auxiliary shading, it is capturing the items which are in shading also(please see the attached picture). So since the issue is fixed now, could you please help what are the changes that I need to make?.


I am attaching the peace of code here in the email below


Please kindly help.

regards,
Sateesh Mandapalli

openlayerMap.addLayer(vectorLayer);

measureLayer = measureControlMap();
openlayerMap.addLayer(measureLayer);
draw = new ol.interaction.Draw({
source: sourceMeasure,
type: /** @type {ol.geom.GeometryType} */ ('Polygon')
});
openlayerMap.addInteraction(draw);
draw.on('drawstart',
function(evt) {
// set sketch
}, this);
draw.on('drawend',
function(evt) {
startLoading();
var extent = evt.feature.getGeometry().getExtent();

var editCageAutoCadIds = '';
var editCabinetAutoCadIds = '';

vectorSource.forEachFeatureIntersectingExtent(extent, function(feature) {

var layer = feature.get('LAYER');
var tagId = feature.get('AUTOCAD_TAG');
if(layer=='CABINET_INFO'){
editCabinetAutoCadIds+=tagId+',';
}else if(layer=='CAGE_INFO'){
editCageAutoCadIds+=tagId+',';
}

});
$('#emrCageAutoCadIds').val(editCageAutoCadIds);
$('#emrCabinetAutoCadIds').val(editCabinetAutoCadIds);
openEMRCageCabinetPopup();
},this);




________________________________
